packages/next/src/index.ts

```diff
@@ -0,0 +1,46 @@
+import NodeClient from '@logto/node';
+import { withIronSessionApiRoute } from 'iron-session/next';
+import { NextApiHandler, NextApiRequest } from 'next';
+
+import NextStorage from './storage';
+import { LogtoNextConfig } from './types';
+
+export default class LogtoClient {
+  private navigateUrl?: string;
+  private storage?: NextStorage;
+  constructor(private readonly config: LogtoNextConfig) {}
+
+  handleSignIn = (
+    redirectUri = `${this.config.baseUrl}${this.config.signInCallbackEndpoint}`
+  ): NextApiHandler =>
+    this.withIronSession(async (request, response) => {
+      const nodeClient = this.createNodeClient(request);
+      await nodeClient.signIn(redirectUri);
+      await this.storage?.save();
+
+      if (this.navigateUrl) {
+        response.redirect(this.navigateUrl);
+      }
+    });
+
+  private createNodeClient(request: NextApiRequest) {
+    this.storage = new NextStorage(request);
+
+    return new NodeClient(this.config, {
+      storage: this.storage,
+      navigate: (url) => {
+        this.navigateUrl = url;
+      },
+    });
+  }
+
+  private withIronSession(handler: NextApiHandler) {
+    return withIronSessionApiRoute(handler, {
+      cookieName: `logto:${this.config.appId}`,
+      password: this.config.cookieSecret,
+      cookieOptions: {
+        secure: this.config.cookieSecure,
+      },
+    });
+  }
+}
```

Comment:
Can we provide hooks instead of class?

Reply:
It's server side so I guess nothing related to React?

Reply:
Nah, this is Next.js SDK, and since Next is a `React framework` that build web app on the node server, it uses almost every concept in React, including JSX, React Dom, etc...
So I would assume the Next.js SDK would be much similar as React SDK

Reply:
Is all been done in server side, but we can still provide a hook like "useLogto", and fetch info from an api endpoint later.

But this class is necessary.